clang/test/SemaSYCL/intel-fpga-loops.cpp

```diff
@@ -229,6 +229,36 @@ void ivdep_dependent() {
   };
 }
 
+template <int A, int B, int C>
```

Comment:
Can you test a non-dependent attribute in a dependent context?  I have a feeling that the transforms running on a non-dependent version will cause the int value to be lost.

Reply:
You're right, int value is lost in this case. Do you have any suggestions why it happens or how to handle this? As I see, it requires more than one-line fix, but unfortunately I have no idea where the error can be hidden.

Reply:
I think the best way about this is to just remove the IntervalValue field from the attribute.  I don't think it is necessary anyway.  It'll simplify a bunch of code, and result in only the Expr needing to be handled everywhere.  It means you need to do a touch more work in CGLoopInfo (call isIntegerConstantExpr() on the expr to get the integer value out), but it simplifies the rest of the patch a bunch.